src/components/AuthForm.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useContext } from "react"
+import { useHistory } from "react-router-dom"
+import { makeStyles } from "@material-ui/core/styles"
+import { InputAdornment, TextField } from "@material-ui/core"
+import LogoutImage from "../assets/logout-image.svg"
+import classes from "./AuthForm.module.css"
+import { AccountCircle } from "@material-ui/icons"
+import PrimaryButton from "../UI/PrimaryButton"
+import IconButton from "@material-ui/core/IconButton"
+import Visibility from "@material-ui/icons/Visibility"
+import VisibilityOff from "@material-ui/icons/VisibilityOff"
+import { AuthContext } from "../store/auth-context"
+// import {RouteComponentProps} from 'react-router-dom'
+
+const useStyles = makeStyles(() => ({
+  logoutField: {
+    width: "30ch",
+  },
+}))
+
+const AuthForm: React.FC = () => {
+  const [showPassword, setShowPassword] = React.useState<boolean>()
+  const [errorEmailMessage, setErrorEmailMessage] = useState<boolean>(false)
+  const [errorPasswordMessage, setErrorPasswordMessage] =
+    useState<boolean>(false)
+  const styles = useStyles()
+  const authCtx = useContext(AuthContext)
+  const { enteredEmail } = authCtx
+  const { enteredPassword } = authCtx
+  const { enterEmail } = authCtx
+  const { enterPassword } = authCtx
+  const { isLogin } = authCtx
+  const { handleIsLogin } = authCtx
+  const { fetchData } = authCtx
+  const { isLoading } = authCtx
```

Comment:
we can rewrite this in the following way 

`const { enteredEmail, enteredPassword, enterEmail, handleIsLogin } = authCtx`